libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -210,40 +210,39 @@ void AC_AttitudeControl::reset_rate_controller_I_terms_smoothly()
 //    trust vector drops below 2*AC_ATTITUDE_THRUST_ERROR_ANGLE. At this point the heading is also corrected.
 
 // Command a Quaternion attitude with feedforward and smoothing
-void AC_AttitudeControl::input_quaternion(Quaternion attitude_desired_quat)
+void AC_AttitudeControl::input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_target)
 {
```

Comment:
```suggestion
void AC_AttitudeControl::input_quaternion(const Quaternion& attitude_desired_quaternion, const Vector3f &angular_vel_target)
{
    // Inputs
    Quaternion attitude_desired_quat = attitude_desired_quaternion;
    Vector3f ang_vel_target = angular_vel_target;
```

Reply:
You are removing an important task of this function.